### PR TITLE
Updating  Pendo Accounts 

### PR DESCRIPTION
**Pull request by nisargpatel-m**

### CODE REVIEW

File: airbyte-integrations/connectors/source-pendo-python/source_pendo_python/source.py [hunk 1]
```diff
@@ -56,29 +57,38 @@ def get_reports(self, config):
...
+            return [obj["id"] for obj in body]
         except requests.exceptions.RequestException as e:
             return False, e
 
     def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         auth = self._get_authenticator(config)
 
+        default_start_date = datetime.now() - timedelta(days=2*365)
+        start_date = config.get("start_date", default_start_date.strftime('%Y-%m-%dT%H:%M:%S'))
+        day_page_size = config.get("day_page_size")
```
Comment (msaffitz) [inline airbyte-integrations/connectors/source-pendo-python/source_pendo_python/source.py:69]: should we add some sane default here?  (previously we had 21 days)

File: airbyte-integrations/connectors/source-pendo-python/source_pendo_python/source.py [hunk 2]
```diff
@@ -56,29 +57,38 @@ def get_reports(self, config):
...
+            PageEvents(start_date=start_date, day_page_size=day_page_size, authenticator=auth),
+            FeatureEvents(start_date=start_date, day_page_size=day_page_size, authenticator=auth),
+            GuideEvents(start_date=start_date, day_page_size=day_page_size, authenticator=auth)
         ]
 
-        # all_reports = self.get_reports(config)
-        # for report in all_reports:
-        #     result.append(ReportResult(report=report, authenticator=auth))
+        all_reports = self.get_reports(config)
+        reports = config.get("reports")
```
Comment (msaffitz) [inline airbyte-integrations/connectors/source-pendo-python/source_pendo_python/source.py:86]: small naming suggestion:  call `reports` something like `report_allowlist` in the config object to reduce potential confusion.

File: airbyte-integrations/connectors/source-pendo-python/source_pendo_python/source.py [hunk 3]
```diff
@@ -56,29 +57,38 @@ def get_reports(self, config):
...
-            GuideEvents(authenticator=auth)
+            PageEvents(start_date=start_date, day_page_size=day_page_size, authenticator=auth),
+            FeatureEvents(start_date=start_date, day_page_size=day_page_size, authenticator=auth),
+            GuideEvents(start_date=start_date, day_page_size=day_page_size, authenticator=auth)
         ]
 
-        # all_reports = self.get_reports(config)
-        # for report in all_reports:
-        #     result.append(ReportResult(report=report, authenticator=auth))
+        all_reports = self.get_reports(config)
```
Comment (msaffitz) [inline airbyte-integrations/connectors/source-pendo-python/source_pendo_python/source.py:85]: no need to call `all_reports` here as you may not use it; you could move  into the else clause for an efficiency gain

File: airbyte-integrations/connectors/source-pendo-python/source_pendo_python/streams.py [hunk 4]
```diff
@@ -42,7 +49,15 @@ def build_schema(self, full_schema, metadata):
...
-                    fields[field] = self.get_valid_field_info(field_type)
+                    if field == "pricingtiersf":
+                        fields[field] = {"type": ["null", "integer", "string"]}
+                    elif field == "accountnumbersf":
+                        fields[field] = {"type": ["null", "string", "number"]}
+                    else:
+                        fields[field] = self.get_valid_field_info(field_type)
+                    # field_type = metadata[key][field]["Type"]
+                    # fields[field] = self.get_valid_field_info(field_type)
+                    # print(f'Field - {field}, {fields[field]}')
```
Comment (msaffitz) [inline airbyte-integrations/connectors/source-pendo-python/source_pendo_python/streams.py:60]: Perhaps worth a small comment here on why we're special casing these

**Review by BillMagnifyIO — APPROVED** [review]

**Pull request merged**